torchao/prototype/mx_formats/mx_linear.py

```diff
@@ -87,13 +97,20 @@ class MXLinear(torch.nn.Linear):
     matmul is emulated since there is no hardware support yet. Activations,
     weights and grads are casted to MX and back to high precision for each
     matmul.
+
+    Input, weight and grad_output can have each their own MX element dtype
+    by passing a tuple of `elem_dtype` to the factory method `from_float`.
     """
 
     @classmethod
     @torch.no_grad()
     def from_float(cls, mod, elem_dtype, block_size):
         mod.__class__ = MXLinear
-        mod.elem_dtype = elem_dtype
+        # Single element dtype passed for input, weight and gradient.
```

Comment:
nit: can we do

```
def from_float(
    ...,
    elem_dtype=torch.float8_e4m3fn,
    elem_dtype_weight_override=None,
    elem_dtype_grad_output_override=None,
    ...
): ...
```

we plan to create a proper config object for this in the future, but for now would be good to keep things simple and avoid mixing types in the API (such as dtype vs tuple)

Reply:
Should I then enforce named argument in `MXLinear.from_float` and `swap_linear_with_mx_linear` for `block_size` and `filter_fn`? And have a default `block_size=32`?

Reply:
sounds reasonable!

Reply:
Just pushed a fix commit with:
```python
def from_float(
        cls,
        mod,
        elem_dtype,
        elem_dtype_weight_override=None,
        elem_dtype_grad_output_override=None,
        *,
        block_size=32,
    ):
```
and similarly for `swap_linear_with_mx_linear`